### DISCUSSION TITLE
Query parameters returning empty string

### DISCUSSION THREAD

**Discussion by samantha-wong** [vercel/next.js#34424] [Help]
Hi, has anyone experienced query returning an empty string? It stopped working for me all of a sudden. I've tried retrieving it using context.query.sId and through router.query.sId. I've also tried using router.isReady, the craziest part is it works fine in localhost. 

<img width="1372" alt="image" src="https://user-images.githubusercontent.com/19571028/154279356-f0a18943-57b7-4bbc-864a-b1e7635e3fae.png">

I'm using version 12.0.3. Any help appreciated! 

Thanks!

**Comment by 2stash** [comment]
@samantha-wong Your 'sId' should be working as your url looks good. I don't think it is related to a nextjs issue.

I could be where you are trying to deploy this. But also with a quick google search it could be related to ad-blockers or some third party program blocking the ability to pass a query in the URL. So I would try looking for those and disabling them.

https://stackoverflow.com/questions/48555048/how-do-i-fix-the-err-blocked-by-client-error-with-youtube-data-api-v3-when-disp

**Reply by samantha-wong** [reply]
Unfortunately this isn't the case, I tried a new repo and it works.

**Reply by 2stash** [reply]
@samantha-wong Hi can you elaborate more? Did you solve your problem with the new repo? I assume you started a new repo and copied over your logic and it started working? Was the new repo the same version number? Or did you do something else? Also what do you mean it was working and all of a sudden it stopped working?

Looking through the open issues I found 2 that are related to losing query parameters. Please read through them and see if any are similar.

https://github.com/vercel/next.js/issues/24483

https://github.com/vercel/next.js/issues/24532

**Reply by samantha-wong** [reply]
I figured it out thanks! Turns out it silently failed because of using relative urls in [middleware](https://nextjs.org/docs/messages/middleware-relative-urls). Just updated our middleware and it's populating again.

Our version was pinned to 12.0.3 and support got pulled without warning... :( - issue started happening a few weeks ago
